MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/SearchVC/SearchViewController.swift

```diff
@@ -0,0 +1,84 @@
+import UIKit
+
+class SearchViewController: UIViewController {
+    
+    let offset: CGFloat = 4
+    
+    var movies: [MovieViewModel]?
+    
+    var tableView: UITableView!
+    var presenter: SearchPresenter!
+    
+    init(presenter: SearchPresenter) {
+        super.init(nibName: nil, bundle: nil)
+        
+        self.presenter = presenter
+        self.presenter.setDelegate(delegate: self)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        buildViews()
+        setupTableView()
+    }
+    
+    private func setupTableView() {
+        tableView.register(MovieCell.self, forCellReuseIdentifier: MovieCell.reuseIdentifier)
+        tableView.dataSource = self
+        tableView.delegate = self
+    }
+    
+}
+
+extension SearchViewController: UITableViewDelegate {
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        1
+    }
+
+}
+
+extension SearchViewController: UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        movies?.count ?? 0
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard
+            let cell = tableView.dequeueReusableCell(
+                withIdentifier: MovieCell.reuseIdentifier,
+                for: indexPath) as? MovieCell,
+            let movie = movies?[indexPath.row]
+        else {
+            return UITableViewCell()
+        }
+        
+        cell.populate(with: movie)
+        cell.selectionStyle = .none
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let cell = tableView.cellForRow(at: indexPath) as? MovieCell else { return }
+        
+        presenter.selectedMovie(withId: cell.movieId)
```

Comment:
you can get a movie id like this

```swift
let id = movies[indexPath.row].id
```

Reply:
[Changed](https://github.com/lihter/Movie-App/pull/41/commits/3383b62f0e45eb7e1f5ba5b650a01beaa73387a1#diff-6c7e4d8b35d5ff6def2dc66860ecfdc7d2f474e451cb72934495033eeeab86a1R70)